### DISCUSSION TITLE
Number of DMX fixtures

### DISCUSSION THREAD

**Discussion by cgiles** [someweisguy/esp_dmx#103] [General]
Hello I was reading your readme, and you wrote in the DMX basics part that "a DMX system consists of one DMX controller and up to 32 DMX fixtures per DMX port.". Shouldn't be 512 dmx fixtures ?

**Comment by Florida-Rob** [comment]
Its 512 addresses, but due to load on the RS485 bus, it’s a max of 32 devices

 

 

From: cgiles ***@***.***> 
Sent: Wednesday, October 4, 2023 9:45 AM
To: someweisguy/esp_dmx ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [someweisguy/esp_dmx] Number of DMX devices (Discussion #103)

 

Hello I was reading your readme, and you wrote in the DMX basics part that "a DMX system consists of one DMX controller and up to 32 DMX fixtures per DMX port.". Shouldn't be 512 dmx fixtures ?

**Comment by someweisguy** [comment]
@riwalker is correct - DMX is theoretically only able to support up to 32 devices on 1 universe. To use more than 32 devices it is required to use multiple universes or a DMX opto-isolator.

In practice, you can sometimes exceed the 32 device capacity but it is not recommended. Exceeding the fixture capacity can lead to erratic and unwanted behavior such as devices no longer responding to DMX commands.

**Comment by markktoo** [comment]
The 512 addresses of DMX are control slots not devices (fixtures). These slots can be for level (intensity) control of a light or for some other purpose such as selecting a color from a color wheel. A fixture can use one or more DMX address slots to be controlled. An RGB LED can use three or four DMX addresses (slots) to be controlled. The 4th for an RGB fixture is if it includes a separate intensity control.

The industry provides DMX/RDM splitters to connect more than the 32 fixtures to a controller. These splitters must support RDM.
